sdk/core/azure-core/src/main/java/com/azure/android/core/util/Base64Util.java

```diff
@@ -83,6 +88,35 @@ public static String encodeToString(byte[] src) {
      * @return The decoded byte array.
      */
     public static byte[] decodeString(String encoded) {
-        return encoded == null ? null : Base64.decode(encoded, Base64.DEFAULT);
+        return encoded == null ? null : base64Wrapper.decode(encoded, Base64.DEFAULT);
+    }
+
+    interface Base64Wrapper {
```

Comment:
Is this planned to be used in the public API? If so, instead of maintaining context in this static helper class should a new overload to each API be added which accepts this interface? That way this class remains thread safe and stateless as expected with a static helper class.

Reply:
This is not planned to be a part of the public API. The reason I wrote this interface was so we wouldn't have to mock the static methods in Android's Base64 class. This is due to a couple of reasons:

- Unfortunately, we cannot use Java's version of this class since it only works with a higher Android version than our supported minimum.
- We cannot access the Android framework's classes in unit tests, which are only accessible when running on the actual operating system on an emulator or physical device. Unit tests run on the local JVM.